src/components/ui/TextField/README.md

```diff
@@ -0,0 +1,136 @@
+
+Where possible, programmatically determine default values to `TextFields` to make it easier for users to complete a form. When providing defaults, make sure those defaults are user-editable in case of an error or false positive.
+
+## Voice & Tone
+
+**Label text** should be short and snappy. Aim for one to three words in total. Don't use conversational labels or any punctuation at the end of the label. ("Name" vs "Tell me your name.")
+
+**Hint text** should use full sentences and can be more verbose. ("Enter your full postal address, including your postal code.")
+
+**Placeholder text** should only be used for examples; not for instructions or extra context. Include “e.g.” before examples and write in sentence case.
+
+## Accessibility
+
+### Use visible labels
+
+A text input should always have a visible label. Screen reader users rely on these to know what to input.
+
+Labels must be visible when an input gains focus, and must be announced to screen readers on focus.
+
+**Never** use a placeholder as a label for an input. These disappear once the user starts typing text and so should only be used for examples to underline a user's understanding of what they need to enter. Use a label to label the input, and use help text to provide supplementary information.
+
+### Only ask questions you really need to ask
+
+Limit the number of inputs (of any type) on a single page. When possible, improve user focus by only asking one question per page and splitting longer forms across multiple pages.
+
+Only ask users for information you really require. There's a mental cost to every additional input you ask a user to fill in ([https://en.wikipedia.org/wiki/Decision_fatigue](https://en.wikipedia.org/wiki/Decision_fatigue)). If the majority of your inputs are optional, you should consider removing those you don't need.
+
+Be very careful asking for sensitive data (titles, gender, ethnicity, age, etc). Collecting this data requires careful privacy and data retention considerations and may make users uncomfortable. It can also be used to inadvertently reinforce stereotypes. Ask yourself why you are collecting data, and, if you can, avoid it entirely.
+
+### Use the correct type
+
+Mobile devices and some assistive technologies adapt the software keyboard to the current task. This helps users with mobility, vision, and cognitive issues to enter information more easily.
+
+If the field is for an email address, for instance, use `type="email"`; this will often enable a software keyboard with an `@` symbol more easily accessible.
+
+### Order labels and help text correctly
+
+Ensure the helper text that appears under an input is read when an assistive technology user stops at an input using ARIA.
+
+### Each input should have a unique ID
+
+Good news: Nautilus does this for you already! If you _do_ supply a custom `id`, `TextField` will detect this and assign things like the `<label>`'s `htmlFor` the correct value.
+
+### Don't be prescriptive about names
```

Comment:
I added this section, but I'm not sure it belongs here. It's possible this is just a personal pet peeve of mine, but if you think it applies it can stay. Fine if not.

Reply:
It absolutely belongs here! (Or, possibly, in a larger Forms component, but let's deal with it then if we want/need to move it up a level.)

Thanks for adding this, it's good guidance.